kartothek/io/dask/_update.py

```diff
@@ -14,6 +15,26 @@
 
 from ._utils import map_delayed
 
+_KTK_HASH_BUCKET = "__KTK_HASH_BUCKET"
+
+
+def _hash_bucket(df: pd.DataFrame, subset: List[str], num_buckets: int):
+    """
+    Categorize each row of `df` based on the data in the columns `subset`
+    into `num_buckets` values. This is based on `pandas.util.hash_pandas_object`
+    """
+
+    if subset is None:
+        subset = df.columns
+    hash_arr = pd.util.hash_pandas_object(df[subset], index=False)
```

Comment:
what determinism guarantees do we have here? is there any host-based / time-based seed or something involved?

Reply:
From my understanding it's deterministic and doesn't involve host based seeds but https://github.com/pandas-dev/pandas/blob/af1585d697aed306ec15c56f524d8481742ba6eb/pandas/core/util/hashing.py#L26

Reply:
I have added a test to address this concern